### DISCUSSION TITLE
MQTT Integration in Homeassistant Limit Node Red Flows

### DISCUSSION THREAD

**Discussion by nady2804** [tbnobody/OpenDTU#677] [Q&A]
Hallo,
Habe jetzt seit längere Zeit die OpenDTU mit Homeassistant Integration am laufen.
Werte werden alle angezeigt.
Das Thema MQTT und Node Red ist jedoch Neuland für mich.
Ich würde gerne Node Red integation vom Homeassistant nutzen um hier das Limit Lastabhängig am Hoymiles zu setzen.
Kann mir hier einer einen einfachen Ansatz erklären bzw den Flow für Node Red erklären. 
Sensoren hab ich bereits alle welche ich benötige im Homeassistant eingebunden jedoch ist mit das mit dem MQTT Limit Werte setzen nicht klar.
Dh mit der Bitte eines Beispiel Node Red Flows.
Ziel wäre es einen Flow zu haben der das Limit am WR in der Abhängigkeit der erzeugbaren Leistung und des Eigenverbrauchs anpassen kann. Dh die maximale Einspeiseleistung zurück ins Netz von 600 Watt sollte nicht überschritten werden. 
Wenn mehr Leistung benötigt wird sollte das Limit aufgehoben werden bei guter Sonneneinstrahlung bei Grundverbrauch eben auf die max 600 Watt zurück ins Netz limitiert werden.

Bin für jede Hilfe dankbar

**Comment by helgeerbe** [comment]
Wenn du eh alles in Home Assistant hast, warum machst du nicht einfach eine Automatisierung in Home Assistant?
Neuer Hoymiles Wert = Aktueller Hoymiles Wert - Exportwert vom PowerMeter + 600 Watt

**Reply by nady2804** [reply]
Ja das thema ist wie ich geschrieben habe eben Neuland für mich, Mir gehts wie ich das in Node Red Flow und Mqtt push realisiere. Würde hier  eben Hilfe benötigen. Die Power Sensoren Entitäten kann ich ja dann anpassen an meine....

**Reply by MrReSc** [reply]
Du brauchst weder Node Red noch musst du etwas explizit via Mqtt pushen. In HA kannst du eine Automation erstellen, die auf die Entität vom entsprechen Limit (z.B. bei mir `number.nordseite_limit_nonpersistent_absolute`) schreibt. Ich würde mich zuerst einmal damit befassen, wie die Automatisierungen funktionieren. Dazu musst du ein Grundverständnis von Logik und Ablaufsteuerungen haben bzw. aufbauen. Dann kriegst du das hin.

**Comment by nady2804** [comment]
Das Grundverständnis von Homeassistant und Automation bzw. Node Red habe ich mir durch zahlreiche Beispiele schon angeeignet. Ich würde es jedoch mit Node Red insofern machen weil das ganze auch etwas komplizierter ist. Ich habe zb 2 WR 1x 800W Ost/West und nich einen Süd mit 800W. Weiters gehe ich davon aus das beide zusammen bei opt. Bedingungen max 1500W liefern können. Dh ich würde zum Beispiel  gerne abhängig der Last bzw. Je nach Einspeise Höhe die WR limitierten. Wobei es mir eigentlich nur wichtig wäre die 600W nicht zu überschreiten. Das Limit aufzuheben ist ja einfach ist die Grundlast (bei mir 200w) und die andere derzeitiger strom Grösser Zb bei >1100W Verbrauch wird das Limit auf 100% gesetzt. Kompliziert wird es jetzt wenn der Stromverbrauch zwischen Grundlast und <1100 watt sich bewegt. Und ich suche eine Lösung die nicht im Sekunden den WR neue Limit Werte übermittelt sondern zb 3 Bereiche unterhalb der 1100W die die Limits setzen aber immer nur das Max 600w im schlechtesten Fall also bei keiner Last eingespeist wird. Und das ganze eben für 2 WR.
